File: Backend/Infrastructure.MyShop/Interfaces/IProductService.cs
using Data.MyShop.Entities;
using Infrastructure.MyShop.Models.DTO.CategoryDTO;
using Infrastructure.MyShop.Models.DTO.ProductDTO;
using Infrastructure.MyShop.Services;
using Microsoft.Extensions.Options;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infrastructure.MyShop.Interfaces
{
    public interface IProductService
    {
        Task<ServiceResponse> GetProductsAsync();
        Task<ServiceResponse> GetProductByNameAsync(string name);
        Task<ServiceResponse> GetProductByIdAsync(int id);
        Task<ServiceResponse> GetProductsByCategoryId(int id);
        Task<ServiceResponse> CreateProductAsync(ProductCreateDTO model);
        //Task<ServiceResponse> GetProductCountAsync();
        //Task<List<ProductImageEntity>> GetProductImagesByProductIdAsync(int id);
        Task<ServiceResponse> EditProductAsync(ProductEditDTO model);
        Task<ServiceResponse> DeleteProductAsync(int id);

    }
}

File: Backend-Java/src/main/java/org/example/Infrastructure/interfaces/IServices/IProductService.java
package org.example.Infrastructure.interfaces.IServices;

import org.example.Infrastructure.dto.productDTO.ProductCreateDTO;
import org.example.Infrastructure.dto.productDTO.ProductEditDTO;
import org.example.Infrastructure.dto.productDTO.ProductItemDTO;

import java.util.List;

public interface IProductService {
    ProductItemDTO create(ProductCreateDTO model);
    List<ProductItemDTO> getAll();
    ProductItemDTO edit(ProductEditDTO model);
    void deleteById(int id);
    ProductItemDTO getById(int id);
}
